crates/cranelift/src/gc/enabled/drc.rs

```diff
@@ -283,8 +283,8 @@ fn emit_gc_raw_alloc(
         .call(gc_alloc_raw_builtin, &[vmctx, kind, ty, size, align]);
 
     let gc_ref = builder.func.dfg.first_result(call_inst);
+    let gc_ref = builder.ins().ireduce(ir::types::I32, gc_ref);
     builder.declare_value_needs_stack_map(gc_ref);
-
```

Comment:
Wait, we should only ever have `i32` values in stack maps, but this will make it include an `i64` that represents a `Result<u32>` IIUC, so this change looks pretty suspect to me. Is this intentional?

Reply:
Oh I thought that the `ireduce` preceding this would take care of that, but is that not the case?

Reply:
Ah whoops I misread this as moving the `ireduce` *after* the needs-stack-map declaration. Carry on!